codegen/cmd/injection-gen/generators/reconciler_reconciler.go

```diff
@@ -367,18 +367,41 @@ func (r *reconcilerImpl) updateStatus(existing *{{.type|raw}}, desired *{{.type|
 			}
 		}
 
+		fullUpdate := false
+		statusUpdate := false
+
+		if !reflect.DeepEqual(existing.GetAnnotations(),desired.GetAnnotations()) {
+			existing.ObjectMeta.Annotations = desired.ObjectMeta.Annotations
+			fullUpdate = true
+		}
+
+		if !reflect.DeepEqual(existing.GetLabels(),desired.GetLabels()) {
```

Comment:
Is `DeepEqual` the right operation? I think users should be allowed to label/annotate downstream resources at their will and not get that reconciled away? If anything, we should ensure that "our" annotations and labels are properly set? 🤔

Reply:
We need to compare `currrent` to `desired` using `relect.DeepDerivative(desired, current)` from `k8s.io/apimachinery/third_party/forked/golang/reflect`. It returns `true` if there are additional values in `current` but none of the values from `desired` have changed.

Reply:
We'll then also need to use `UnionMaps` instead of straight up overriding things.